packages/loot-core/migrations/1699454123000_reports.sql

```diff
@@ -0,0 +1,27 @@
+BEGIN TRANSACTION;
+
+CREATE TABLE reports
+  (id TEXT PRIMARY KEY,
+   name TEXT,
+   start TEXT,
+   end TEXT, 
+   mode TEXT DEFAULT 'total',
+   groupBy TEXT DEFAULT 'Category',
+   balanceType TEXT DEFAULT 'Expense',
+   interval TEXT DEFAULT 'Monthly',
+   empty INTEGER DEFAULT 0,
+   hidden INTEGER DEFAULT 0,
+   uncat INTEGER DEFAULT 0,
+   selectedCategories TEXT,
+   graphType TEXT DEFAULT 'BarGraph',
+   viewLegend INTEGER DEFAULT 0,
+   viewSummary INTEGER DEFAULT 0,
+   viewLabels INTEGER DEFAULT 0,
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏I'm a bit torn about these. From one perspective.. it makes sense to store them in the database. But then the user experience might be a bit jarring.

Let me explain.

Lets say you have 5x widgets with various `viewSummary` states. You open up one widget - it opens summary side panel. Then you open a different - it does NOT open summary side panel. Which to me feels a bit annoying. I would expect to have it open everywhere if I had it open before. So to use userPrefs for the global visibility of the summary side panel instead of a per-widget config.

WDYT?

Reply:
Going to go with global variables I think.